lib/srv/db/mysql/engine.go

```diff
@@ -191,6 +197,26 @@ func (e *Engine) connect(ctx context.Context, sessionCtx *common.Session) (*clie
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
+		// workaround issue generating ephemeral certs for secure connections
+		// by creating a TLS connection to the cloud proxy port (3307) and
+		// overridding mysql client's connection. mysql on 3306 does not trust
+		// the ephemeral cert's CA but cloud proxy does.
+		if sessionCtx.Database.GetTLS().Mode != types.DatabaseTLSMode_INSECURE {
```

Comment:
Insecure mode affects server certificate validation, I don't think it's really relevant for client certificates. Following up on my comment above, I think we should rather check if the instance requires client certificate and use that as an indicator that we should generate a client cert and connect to the proxy port.